explorer-api/src/ping/http.rs

```diff
@@ -1,19 +1,103 @@
+use std::collections::HashMap;
+use std::net::ToSocketAddrs;
+use std::time::Duration;
+
 use rocket::serde::json::Json;
-use rocket::Route;
-use serde::Deserialize;
-use serde::Serialize;
+use rocket::{Route, State};
+
+use crate::ping::models::PingResponse;
+use crate::state::ExplorerApiStateContext;
+
+const CONNECTION_TIMEOUT_SECONDS: u64 = 10;
 
 pub fn ping_make_default_routes() -> Vec<Route> {
     routes_with_openapi![index]
 }
 
-#[derive(Deserialize, Serialize, JsonSchema)]
-pub(crate) struct PingResponse {
-    response_time: u32,
+#[openapi(tag = "ping")]
+#[get("/<pubkey>")]
+pub(crate) async fn index(
+    pubkey: &str,
+    state: &State<ExplorerApiStateContext>,
+) -> Option<Json<PingResponse>> {
+    match state.inner.ping_cache.clone().get(pubkey.to_string()).await {
+        Some(cache_value) => {
+            trace!("Returning cached value for {}", pubkey);
+            Some(Json(PingResponse {
+                ports: cache_value.ports,
+            }))
+        }
+        None => {
+            trace!("No cache value for {}", pubkey);
+
+            match state.inner.get_mix_node(pubkey).await {
+                Some(bond) => {
+                    let mut ports: HashMap<u16, bool> = HashMap::new();
```

Comment:
This test kinda looks like it wanted to live in a separate function

Reply:
Yup, moved into separate function